### PR TITLE
Refresh token logic

### PR DESCRIPTION
**Pull request by fortamt**
develop

## GitHub Board

**Issue link**

[Refresh Token logic #549](https://github.com/ita-social-projects/dokazovi-be/issues/549 )

**Story link**

[-](- )

## Code reviewers
- [ ] @teaFunny 

## Summary of issue

- [x]  create /refreshtoken endpoint
- [x] add refresh token logic

## Summary of change

- [x]  created /refreshtoken endpoint
- [x]  added refresh token repository
- [x] added table for refresh tokens(flyway migrations)
- [x] added RefreshTokenService.class
- [x] added Tests for all
- [x] added links for enviroment variables

### CODE REVIEW

File: src/main/java/com/softserveinc/dokazovi/security/RefreshTokenService.java
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/security/RefreshTokenService.java:39]: Does something changes after 
`refreshTokenRepository.save(refreshToken);`
Why are you reassigning refreshToken variable?

File: src/main/java/com/softserveinc/dokazovi/security/RefreshTokenService.java
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/security/RefreshTokenService.java:53]: Please remove empty lines

File: src/main/java/com/softserveinc/dokazovi/security/TokenProvider.java
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/security/TokenProvider.java:50]: Please remove empty lines

File: src/main/resources/application.properties
Comment (V-Kaidash) [inline src/main/resources/application.properties:44]: What happened if these two parameters not set?
Does it make sense to set some default values?

File: src/main/java/com/softserveinc/dokazovi/config/AppProperties.java
Comment (V-Kaidash) [inline src/main/java/com/softserveinc/dokazovi/config/AppProperties.java:15]: Please check your IDE settings for indentation and tabs and adjust your code accordingly. Also check other files you have edited. IDE settings should be something like this 
image

File: src/test/java/com/softserveinc/dokazovi/security/RefreshTokenServiceTest.java
Comment (V-Kaidash) [inline src/test/java/com/softserveinc/dokazovi/security/RefreshTokenServiceTest.java:84]: Please remove empty lines & fix indentation (see IDE settings above)

**Review by V-Kaidash — CHANGES_REQUESTED** [review]

**Review by V-Kaidash — APPROVED** [review]
Refresh token proof/demo
https://drive.google.com/file/d/1KBhr29pPqhQ5N0Asw3RhYo1kZlmL02fS/view?usp=sharing

All looks good for me.
Let's wait on the final review from Alina before merging.

**Review by teaFunny — APPROVED** [review]

**Pull request merged**
